internal/provider/resource_tfe_registry_module_test.go

```diff
@@ -803,6 +805,107 @@ func TestAccTFERegistryModule_invalidWithRegistryNameAndNoModuleProvider(t *test
 		},
 	})
 }
+
+func TestAccTFERegistryModule_moduleProviderDestroyedOnDelete(t *testing.T) {
+	registryModule := &tfe.RegistryModule{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	orgName := fmt.Sprintf("tst-terraform-%d", rInt)
+
+	expectedRegistryModuleAttributes := &tfe.RegistryModule{
+		Organization: &tfe.Organization{Name: orgName},
+		Name:         "foo-aws-mod",
+		Provider:     "aws",
+		RegistryName: tfe.PrivateRegistry,
+		Namespace:    orgName,
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFERegistryModuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFERegistryModule_moduleProviderDestroyedOnDelete(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFERegistryModuleExists("tfe_registry_module.foo",
+						tfe.RegistryModuleID{
+							Organization: orgName,
+							Provider:     expectedRegistryModuleAttributes.Provider,
+							Name:         expectedRegistryModuleAttributes.Name,
+							RegistryName: expectedRegistryModuleAttributes.RegistryName,
+							Namespace:    expectedRegistryModuleAttributes.Namespace,
+						}, registryModule),
+					testAccCheckTFERegistryModuleExists("tfe_registry_module.bar",
+						tfe.RegistryModuleID{
+							Organization: orgName,
+							Provider:     expectedRegistryModuleAttributes.Provider,
+							Name:         expectedRegistryModuleAttributes.Name,
+							RegistryName: expectedRegistryModuleAttributes.RegistryName,
+							Namespace:    expectedRegistryModuleAttributes.Namespace,
+						}, registryModule),
+					testAccCheckTFERegistryModuleExists("tfe_registry_module.baz",
+						tfe.RegistryModuleID{
+							Organization: orgName,
+							Provider:     expectedRegistryModuleAttributes.Provider,
+							Name:         expectedRegistryModuleAttributes.Name,
+							RegistryName: expectedRegistryModuleAttributes.RegistryName,
+							Namespace:    expectedRegistryModuleAttributes.Namespace,
+						}, registryModule),
+					testAccTFERegistryModule_deleteModuleProvider("tfe_registry_module.foo",
```

Comment:
this check function seems to delete the registry module as opposed to checking for an expected condition, can you say more about this check?

Reply:
Yep, I interpreted the language here to mean I can use `Check` to make a custom API call. In this case, I am testing the `DeleteProvider` endpoint. Realizing `check the status of things` might only mean existence or not.
```
// Check is called after the Config is applied. Use this step to
// make your own API calls to check the status of things, and to
// inspect the format of the ResourceState itself.
```